packages/frontend/core/src/hooks/affine/use-sidebar-drag.ts

```diff
@@ -0,0 +1,128 @@
+import { toast } from '@affine/component';
+import type { DraggableTitleCellData } from '@affine/component/page-list';
+import { useAFFiNEI18N } from '@affine/i18n/hooks';
+import type { DragEndEvent, UniqueIdentifier } from '@dnd-kit/core';
+import { usePageMetaHelper } from '@toeverything/hooks/use-block-suite-page-meta';
+import { useCallback } from 'react';
+
+import { useCurrentWorkspace } from '../current/use-current-workspace';
+import { useBlockSuiteMetaHelper } from './use-block-suite-meta-helper';
+import { useTrashModalHelper } from './use-trash-modal-helper';
+
+// Unique droppable IDs
+export enum DropPrefix {
+  SidebarCollections = 'sidebar-collections-',
+  SidebarTrash = 'sidebar-trash',
+  SidebarFavorites = 'sidebar-favorites',
+}
+
+export enum DragPrefix {
+  PageListItem = 'page-list-item-title-',
+  FavouriteListItem = 'favourite-list-item-',
+  CollectionListItem = 'collection-list-item-',
+  CollectionListPageItem = 'collection-list-page-item-',
+}
```

Comment:
May not need enum this way. You are using enum as a key -> string mapping, but enum's value should not be used plain strings.
In this case it might be even better to use plain const objects.

Also, better to expose function like `function getDragItemId(type: 'collection' | 'page', id: string): string;` instead.